### DISCUSSION TITLE
How to discover services in a network without a border router?

### DISCUSSION THREAD

**Discussion by souponaroom** [openthread/openthread#8880] [Q&A] [answered]
Hello, I am trying to create a network of wireless sensors and peripherals using thread. Each device can have different capabilities, so there needs to be a way to register their identities in a unique way for each node.

I have found DNS-SD and SRP, but a lot of the documentation seems geared towards usage with a border router. My network does not have one, but will still have various modules added and removed at random intervals.

Since mesh-local EID's seem random, what's the best way to give each device a unique identity?

**Comment by wgtdkp** [comment, marked as answer]
Hey, SRP services are registered to a SRP server and you don't really need a border router to enable the SRP server - simply build one of your device with cmake option `-DOT_SRP_SERVER=ON -DOT_ECDSA=ON`.

Note that each SRP service typically costs around 200 - 300 bytes memory, so you may want to choose a "powerful" device to be the SRP server and enables `OT_EXTERNAL_HEAP` if there are hundreds of devices in your network.

**Reply by souponaroom** [reply]
Thanks for the quick response! Follow up question: what's the difference between SRP and DNS-SD? To me, someone who's never used either, they sound like the same thing.

**Reply by EskoDijk** [reply]
SRP is a registration protocol as defined by https://datatracker.ietf.org/doc/html/draft-ietf-dnssd-srp . It makes use of the DNS-SD syntax and semantics for service descriptions. The intro of this doc also explains a bit more.  The querying of registered services is done using DNS queries, in combination with again the DNS-SD syntax.
